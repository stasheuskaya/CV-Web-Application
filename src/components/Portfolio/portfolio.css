.scrollPortfolio {
    display: flex;
    flex-wrap: nowrap;
    -webkit-overflow-scrolling: touch;
    -ms-overflow-style: -ms-autohiding-scrollbar;
}

.portfolioItem {
    width: 308px;
    height: 190px;
    border: 1px solid var(--grey-color-alto);
    flex: 0 0 auto; 
    position: relative;
}

.portfolioDescription {
    position: absolute;
    top: 0;
    opacity: 0;
    left: 0px;
    height: 0px;
    padding-left: 15px;
    padding-right: 15px;
    transition: all 0.3s ease-out;
    background-color: white;
}

.portfolioDescription:hover {
    opacity: 100%;
}
.portfolioItem:hover .portfolioDescription {
    height: 190px;
}

.portfolioTextTitle {
    margin-top: 15px;
}

.portfolioFilter {
    display: flex;
    height: 20px;
    padding-bottom: 20px;
}

.linkName {
    position: absolute;
    left: 55px;
}

.navigationPanel.compact .linkName {
    display: none;
}

.linkName:hover {
    color: var(--green-color-highlight-shamrock);
}

.filterLinks {
    padding-right: 5px;
}

.filterLinks:hover, .filterLinks.highlight {
    color: var(--green-color-highlight-shamrock);
}

.portfolioItem h1, .portfolioItem a {
    color: var(--green-color-highlight-shamrock);
}

.portfolioItem h2, .portfolioItem h1{
    padding-bottom: 10px;
}
