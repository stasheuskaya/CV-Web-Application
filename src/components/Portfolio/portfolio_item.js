import UI_one from "../../assets/images/UI_one.png";
import UI_two from "../../assets/images/UI_two.png";

export default function PortfolioItem (props) {
    return (
        <div className="portfolioItem">      
            <img 
                src={props.type === "ui" ? UI_one : UI_two} 
                className="portfolioImage" 
                alt="userAvatar"/>
            <div className='portfolioDescription'>
                <h1 className="portfolioTextTitle">
                    {props.title}
                </h1>
                <h2>{props.text}</h2>
                <a href="/inner">View resource</a>
            </div>
        </div>
    )
}
