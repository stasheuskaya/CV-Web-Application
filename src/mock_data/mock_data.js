export const expertisesData = [
  {
      date: '2021-2022', 
      info: {
        company: 'EPAM Systems inc',
        job: 'FrontEnd Developer/Engineer',
        description: 'Develop various web application, including the a Capstone project "CV Web application".  CV web application is a personal, professional CV web application that allows users to have a working prototype to share with audiences to introduce themselves.'
      }
  },
  {
      date: '2019-2021', 
      info: {
        company: 'Selfemployee',
        job: 'Mobile Software Testing Engineer  / QA',
        description: 'MyBudget mobile application (iOS). Set a budget and keep it with real time budget stats.'
      }
  },
  {
    date: '2017-2018', 
    info: {
      company: 'EPAM Systems inc',
      job: 'Mobile Software Testing Engineer  / QA',
      description: 'Ecommerce mobile application (iOS, Android). Global specialist in energy management and automation, with global presence in over 100 countries. It provides integrated efficiency solutions, combining energy, automation and software.'
    }
  },
  {
    date: '2015-2016', 
    info: {
      company: 'EPAM Systems inc',
      job: 'Software Testing Engineer',
      description: 'Ecommerce mobile application (iOS, Android) online-fashion outlet, with operations across Spain, Italy, Brazil and Mexico.'
    }
  },
  {
      date: '2014-2015', 
      info: {
        company: 'EPAM Systems inc',
        job: 'Software Testing Engineer',
        description: 'SaaS based Supply Chain and Industry platform, provides a set of Internet services for the automotive and manufacturing industry, with global presents in over 70 countries.'
      }
    }
]

export const feedbackData = [
  {feedback: '"Yulia is an incredibly talented, hard-working individual. I would recommend her to work in any area she pursues. She volunteered under me and also worked for a school I was on the board for. In both circumstances, paid and unpaid, she was dedicated, faithful, consistent, honest, kind, loving, full of good ideas, took initiative, and easy to work with."', 
  reporter: { photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg', 
  name: 'Allyse Tanovitz, YVCS School board member', 
  citeUrl: 'https://yvcschool.org' } 
  }, 

  {feedback: '"Yuliya was a valuable member of our team, and her job and attitude helped us to finish project successfully. Yuliya proved herself not only as a good QA engineer, but also as a consistent and reliable person, it was easy and comfortable to work with Yuliya."'
  , 
  reporter: 
  { photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg', 
  name: 'Marina Tretjakova, QA Test Lead, EPAM Systems', 
  citeUrl: 'www.epam.com' } } 
]

export const timeLineData = [
{
  "date": 2022,
  "title": "‘Front-end Development‘ technology bootcamp",
  "text": "‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, "
},
{
  "date": 2015,
  "title": "‘Mobile testing’ training",
  "text": "‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK)."
},
{
  "date": 2014,
  "title": "‘Software Testing introduction‘ training",
  "text": "‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat)."
},
{
  "date": 2008,
  "title": "Bachelor in Business Activities",
  "text": " ’Institute of Entrepreneurial Activity’, Belarus."
}
]

export const portfolioData = [
{
  "type": "ui",
  "title": "‘Front-end Development‘ technology bootcamp",
  "text": "‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, "
},
{
  "type": "code",
  "title": "‘Mobile testing’ training",
  "text": "‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK)."
},
{
  "type": "ui",
  "title": "‘Software Testing introduction‘ training",
  "text": "‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat)."
},
{
  "type": "code",
  "title": "Bachelor in Business Activities",
  "text": " ’Institute of Entrepreneurial Activity’, Belarus."
}
]

export const skillsData = [
{
  "name": "React",
  "range": 10
},
{
  "name": "JS",
  "range": 100
}
]