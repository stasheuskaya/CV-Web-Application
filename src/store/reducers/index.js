import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;