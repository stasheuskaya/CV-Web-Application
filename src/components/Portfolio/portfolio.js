import './portfolio.css';
import PortfolioItem from './portfolio_item';
import { useState } from 'react';

const ALL = 'all'
const UI = 'ui'
const CODE = 'code'

export default function Portfolio(props) {
    const [filter, setFilter] = useState(ALL);
    const clickListener = (value) => () => {
        setFilter(value);
    }

    const linkState = (type) => {
        if (filter === type) {
            return " highlight"
        } else {
            return ""
        }
    }

    return (
        <>
            <div className="portfolioFilter">
                <div className={`filterLinks${linkState(ALL)}`} onClick={clickListener(ALL)}>All /</div>
                <div className={`filterLinks${linkState(UI)}`} onClick={clickListener(UI)}>UI /</div>
                <div className={`filterLinks${linkState(CODE)}`} onClick={clickListener(CODE)}>Code</div>
            </div>
            <div className="scrollPortfolio">
                {props.data
                .filter(item => filter === ALL || item.type === filter)
                .map(item => 
                    <PortfolioItem 
                        type={item.type} 
                        title={item.title}
                        text={item.text}
                        key={item.title}>
                    </PortfolioItem>)}
            </div>
        </>
    )
}

