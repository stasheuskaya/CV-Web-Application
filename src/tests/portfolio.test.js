import renderer from "react-test-renderer";
import Portfolio from "../components/Portfolio/portfolio";
import { Provider } from 'react-redux';
import { store } from "../store/store"
import { portfolioData } from "../mock_data/mock_data"

describe('Portfolio Page', () => {
    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
                <Portfolio data={portfolioData}/>
			</Provider>)).toMatchSnapshot();
	});
});