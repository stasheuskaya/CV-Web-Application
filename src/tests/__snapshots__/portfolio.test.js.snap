// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Page renders content 1`] = `
Array [
  <div
    className="portfolioFilter"
  >
    <div
      className="filterLinks highlight"
      onClick={[Function]}
    >
      All /
    </div>
    <div
      className="filterLinks"
      onClick={[Function]}
    >
      UI /
    </div>
    <div
      className="filterLinks"
      onClick={[Function]}
    >
      Code
    </div>
  </div>,
  <div
    className="scrollPortfolio"
  >
    <div
      className="portfolioItem"
    >
      <img
        alt="userAvatar"
        className="portfolioImage"
        src="UI_one.png"
      />
      <div
        className="portfolioDescription"
      >
        <h1
          className="portfolioTextTitle"
        >
          ‘Front-end Development‘ technology bootcamp
        </h1>
        <h2>
          ‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, 
        </h2>
        <a
          href="/inner"
        >
          View resource
        </a>
      </div>
    </div>
    <div
      className="portfolioItem"
    >
      <img
        alt="userAvatar"
        className="portfolioImage"
        src="UI_two.png"
      />
      <div
        className="portfolioDescription"
      >
        <h1
          className="portfolioTextTitle"
        >
          ‘Mobile testing’ training
        </h1>
        <h2>
          ‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK).
        </h2>
        <a
          href="/inner"
        >
          View resource
        </a>
      </div>
    </div>
    <div
      className="portfolioItem"
    >
      <img
        alt="userAvatar"
        className="portfolioImage"
        src="UI_one.png"
      />
      <div
        className="portfolioDescription"
      >
        <h1
          className="portfolioTextTitle"
        >
          ‘Software Testing introduction‘ training
        </h1>
        <h2>
          ‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat).
        </h2>
        <a
          href="/inner"
        >
          View resource
        </a>
      </div>
    </div>
    <div
      className="portfolioItem"
    >
      <img
        alt="userAvatar"
        className="portfolioImage"
        src="UI_two.png"
      />
      <div
        className="portfolioDescription"
      >
        <h1
          className="portfolioTextTitle"
        >
          Bachelor in Business Activities
        </h1>
        <h2>
           ’Institute of Entrepreneurial Activity’, Belarus.
        </h2>
        <a
          href="/inner"
        >
          View resource
        </a>
      </div>
    </div>
  </div>,
]
`;
